src/main/java/app/linguistai/bmvp/service/gamification/StoreService.java

```diff
@@ -156,50 +215,73 @@ public RUserItem purchaseUserItem(String email, UUID storeItemId) throws Excepti
     }
 
     @Transactional
-    public RUserItem decreaseUserItemQuantity(String email, UUID storeItemId) throws Exception {
+    public RUserItem decreaseUserItemQuantity(String email, String type) throws Exception {
         try {
             User user = accountRepository.findUserByEmail(email)
                     .orElseThrow(() -> new NotFoundException(User.class.getSimpleName(), true));
 
-            StoreItem storeItem = storeItemRepository.findById(storeItemId)
+            StoreItem storeItem = storeItemRepository.findByType(type)
                     .orElseThrow(() -> new NotFoundException(StoreItem.class.getSimpleName(), true));
 
             UserItem userItem = userItemRepository.findByUserAndStoreItem(user, storeItem)
                     .orElseThrow(() -> new NotFoundException(UserItem.class.getSimpleName(), true));
 
-            if (userItem.getQuantity() <= 0) {
-                throw new InsufficientUserItemQuantityException();
-            }
-
-            // Consume one unit of the user item
-            userItem.setQuantity(userItem.getQuantity() - 1);
-
-            if (userItem.getQuantity() <= 0) {
-                // If quantity becomes 0 or negative, delete the user item
-                userItemRepository.delete(userItem);
-            } else {
-                // Otherwise, save the updated user item
-                userItemRepository.save(userItem);
-            }
+            consumeUserItem(userItem);
```

Comment:
This could be more readable with the `this.` prefix, but since we are now prioritizing feature velocity, we can sacrifice some maintainability and readability. (just left this as a note for future reference)